gbpservice/neutron/nfp/configurator/api/v1/controllers/controller.py

```diff
@@ -0,0 +1,155 @@
+            msg = ("NOTIFICATION_DATA sent to config_agent %s"
+                   % notification_data)
+            LOG.info(msg)
+            return notification_data
+        except Exception as err:
+            msg = ("Failed to get notification_data  %s."
+                   % str(err).capitalize())
+            LOG.error(msg)
+            return json.dumps({'ERROR': msg})
+
+    @expose(method='POST', content_type='application/json')
+    def post(self, **body):
+        """post method REST server.
+        cast RPC according to the post request by using
+        object of RPCClient"""
+        try:
+            body = None
+            if request.is_body_readable:
+                body = request.json_body
+
+            method = getattr(self.rpcclient, '%s' % self.module_name)
+            method(body)
+            msg = ("Successfully served HTTP request %s" % self.module_name)
+            LOG.info(msg)
+            return json.dumps({'SUCCESS': self.module_name})
+        except Exception as err:
+            msg = ("Failed to serve HTTP post request %s %s."
+                   % (self.module_name, str(err).capitalize()))
+            LOG.error(msg)
+            return json.dumps({'ERROR': msg})
+
+    @expose(method='PUT', content_type='application/json')
+    def put(self, **body):
+        """put method REST server.
+        cast RPC according to the put request by using
+        object of RPCClient"""
+        try:
+            body = None
+            if request.is_body_readable:
+                body = request.json_body
+
+            method = getattr(self.rpcclient, '%s' % self.module_name)
+            method(body)
+            msg = ("Successfully served HTTP request %s" % self.module_name)
+            LOG.info(msg)
+            return json.dumps({'SUCCESS': self.module_name})
+
+        except Exception as err:
+            msg = ("Failed to serve HTTP put request %s %s."
+                   % (self.module_name, str(err).capitalize()))
+            LOG.error(msg)
+            return json.dumps({'ERROR': msg})
+
+
+"""This class make RPC call/cast on behalf of controller class
+    according to the HTTP request coming from config-agent."""
+
+
+class RPCClient(object):
+
+    API_VERSION = '1.0'
+
+    def __init__(self, topic, host):
+
+        self.topic = topic
+        self.host = host
+        target = oslo_messaging.Target(
+            topic=self.topic,
+            version=self.API_VERSION)
+        n_rpc.init(cfg.CONF)
+        self.client = n_rpc.get_client(target)
+
+    def get_notifications(self):
+        """ make rpc call 'get_notifications' """
+        cctxt = self.client.prepare(server=self.host)
+        return cctxt.call(self, 'get_notifications')
+
+    def create_network_function_device_config(self, request_data):
+        """ make rpc cast 'create_network_function_device_config' """
+        cctxt = self.client.prepare(server=self.host)
```

Comment:
Except for get_notifications() method, all other methods have same code. Keep it in same function and invoke it here. Only method name needs to be passed.